addons/event/tests/test_event_mail_schedule.py

```diff
@@ -483,6 +484,52 @@ def test_event_mail_schedule_on_subscription_async(self):
                 'subject': f'Confirmation for {test_event.name}',
             })
 
+    @mute_logger('odoo.addons.base.models.ir_model', 'odoo.models')
+    @users('user_eventmanager')
+    def test_event_mail_schedule_on_subscription_limit(self):
+        """ Schedule should not run on all available attendees, as it may have
+        a big waiting list. Better do it by chunks, which is what is tested. """
+        test_event = self.test_event.with_env(self.env)
+        after_sub_scheduler = test_event.event_mail_ids.filtered(lambda s: s.interval_type == 'after_sub' and s.interval_unit == 'now')
+        cron = self.env.ref('event.event_mail_scheduler')
+
+        original_exec = EventMailScheduler._execute_after_subscription
+        def _patched_execute_after_subscription(self, *args, **kwargs):
+            return original_exec(self, registration_limit=3)
+
+        email_total = 0
+        for registration_count in [2, 3, 5]:
```

Comment:
Mmh, wondering if it would not be better to reset records / counts in-between each batch rather than playing with a global count, as currently the test success is depending on the order of your counts here, it would not work if 5 was first I think.